hypertrace-trace-enricher/trace-reader/src/main/java/org/hypertrace/trace/accessor/entities/DefaultTraceEntityAccessor.java

```diff
@@ -69,16 +69,17 @@ private boolean isExcludedEntityType(final EntityType entityType) {
   }
 
   private void writeEntityIfExists(EntityType entityType, StructuredTrace trace, Event span) {
-    Optional<Entity> entityOptional = this.buildEntity(entityType, trace, span);
-    if (entityOptional.isEmpty()) {
-      return;
-    }
-    this.entityDataClient.createOrUpdateEntityEventually(
-        this.traceAttributeReader.getRequestContext(span),
-        entityOptional.get(),
-        this.buildUpsertCondition(entityType, trace, span)
-            .orElse(UpsertCondition.getDefaultInstance()),
-        this.writeThrottleDuration);
+    this.buildEntity(entityType, trace, span)
+        .map(
+            entity -> {
+              this.entityDataClient.createOrUpdateEntityEventually(
+                  this.traceAttributeReader.getRequestContext(span),
+                  entity,
+                  this.buildUpsertCondition(entityType, trace, span)
+                      .orElse(UpsertCondition.getDefaultInstance()),
+                  this.writeThrottleDuration);
+              return null;
```

Comment:
This looks like a map for side effects (sorry I think I used the word map missing nothing is returned). Instead, use `.ifPresent`